### ISSUE TITLE
Bug in the pycbc_banksim_collect_results file created by workflow

### ISSUE THREAD

**Issue by divyajyoti09** [gwastro/pycbc#4700]
Upon launching `pycbc_make_banksim --conf config.ini` through workflow, the workflow generator automatically creates files which are placed in the scripts folder and executed. The file `scripts/pycbc_banksim_collect_results` has a bug. It is not able to read the bank files properly. It throws the following error:
```
Traceback (most recent call last):
  File "/home/divyajyoti.nln/ENIGMA/FF_calc/scripts/inj_example/scripts/pycbc_banksim_collect_results", line 29, in <module>
    indoc = utils.load_filename(row['bank'], False,
  File "/home/divyajyoti.nln/anaconda3/envs/pycbc_ESIGMA_8Apr24/lib/python3.9/site-packages/ligo/lw/utils/__init__.py", line 427, in load_filename
    with open(filename, "rb") as fileobj:
FileNotFoundError: [Errno 2] No such file or directory: b"b'bank/bank343.xml.gz'"
```
It is reading the match*.dat files which have rows in the following format
```
0.97231 b'bank/bank330.xml.gz' 49 b'injection/injection13.xml.gz' 0 18023862.57156
```

**Comment by divyajyoti09** [comment]
If I modify the file from 
```
if row['bank'] not in btables:
        indoc = utils.load_filename(row['bank'], False,
                                    contenthandler=LIGOLWContentHandler)
```
to 
```
if row['bank'] not in btables:
        indoc = utils.load_filename(eval(row['bank']).decode('utf-8'), False,
                                    contenthandler=LIGOLWContentHandler)
```
then I get another error:
```
Traceback (most recent call last):
  File "/home/divyajyoti.nln/ENIGMA/FF_calc/scripts/inj_example/scripts/pycbc_banksim_collect_results", line 30, in <module>
    btables[row['bank']] = table.get_table(indoc, "sngl_inspiral") 
AttributeError: module 'ligo.lw.table' has no attribute 'get_table'
```

**Comment by prayush** [comment]
@divyajyoti09 is `pycbc_banksim_collect_results` trying to read in a bank file that doesn't exist at all (i.e. does `bank/bank343.xml.gz` exist to begin with)?

**Comment by divyajyoti09** [comment]
@prayush The file exists, but the way its name is stored in the `match*.dat` files is with an extra `b` at the beginning, and while being imported, another `b` gets added, hence the first error, which contains 2 b's in the string. 

I modified the code to get rid of the 2 b's (which essentially converts it to the line of code you have pasted above), and runs that line of code successfully. The following error is in the next line of code (sorry I should've mentioned this in my earlier message, I've updated the message now) which is 
```
btables[row['bank']] = table.get_table(indoc, "sngl_inspiral")
```

**Comment by prayush** [comment]
This should be fixed by #4707
